### PR TITLE
Yojahuang final project proposal

### PR DESCRIPTION
**Pull request by Yojahuang**
---
A good PR should include the following parts:

- [x] A clear title
- [x] A meaningful message for PR, as well as its commits
- [x] from your specific branch (***not master***) merging to teacher's `HW` branch (***not master***, again)
- [x] excluding any irrelevant files, such as binaries, text files, or dot files
- [x] passing tests/CI

### CODE REVIEW

File: project/Yojahuang/README.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+There are many libraries which can solve spatial searching querys. But their performance may be bounded when facing larger scale of dataset and queries.
+
+R tree search engine is a C++ library that aims to provide an efficient and scalable spatial search engine which can serve as a building block of map application.
+
+## Problem to Solve
+
+When we talked about map application, often time we'll have queries like “return all buildings in this area”, “find 1000 closest gas stations to this point”, and returning results within milliseconds even when searching millions of objects.
+
+The goal of R tree search engine is to provide a high performance search engine to solve these classical problems like "K nearest neighbors" and "range query".
```
Comment (yungyuc) [inline project/Yojahuang/README.md:15]: Please include a word or two for how you will evaluation the performance of the tree.

File: project/Yojahuang/README.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+## Prospective Users
+
+This project aim to provide developers that want to build their own map service a cornerstone.
+
+## System Architecture
+
+In python, mostly of the outputs and inputs are given through dictionary.
```
Comment (yungyuc) [inline project/Yojahuang/README.md:23]: For Python API design, please take a look at SciPy KDTree.

File: project/Yojahuang/README.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+In C++, the library will provide class as interface of the library, especially output.
```
Comment (yungyuc) [inline project/Yojahuang/README.md:25]: When it comes to performance, containers for the output are tricky to design.

File: project/Yojahuang/README.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+- Search existing R tree module, tried to run benchmark for them.
+- Check how those libraries provide their interface.
+
+### Week 2 (4/17)
+
+- Won't be doing anything due to I need to participate a hackathon called ETHTaipei
+
+### Week 3 (4/24)
+
+- Plan out the prototype interface, and also the UML graph.
```
Comment (yungyuc) [inline project/Yojahuang/README.md:61]: Use UML if you really like it.  In practice it is not very useful.

File: project/Yojahuang/README.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+- Won't be doing anything due to I need to participate a hackathon called ETHTaipei
+
+### Week 3 (4/24)
+
+- Plan out the prototype interface, and also the UML graph. 
+- Write c++ classes that state out the member functions and variables.
+ 
+### Week 4 (5/1)
+
+- Produce test dataset for essential features.
```
Comment (yungyuc) [inline project/Yojahuang/README.md:66]: It's better to prepare the data for testing in the very beginning.

File: project/Yojahuang/README.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+- Developing the essential features for the library.
+
+### Week 5 (5/8)
+
+- Refactor.
+
+### Week 6 (5/15)
+
+- Optimize the performance.
```
Comment (yungyuc) [inline project/Yojahuang/README.md:75]: You won't have time to modify code for speed if optimization is schedule this late.

File: project/Yojahuang/README.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+### Week 7 (5/22)
+
+- Complete API documentation for the library.
+
+### Week 8 (5/29)
+
+- Prepare the slide for upcoming presentation
```
Comment (yungyuc) [inline project/Yojahuang/README.md:83]: You may consider to use the slides as documentation.  In other words, to develop slides as you document.

File: project/README.rst [hunk 8]
```diff
@@ -68,6 +68,9 @@ Project List by Account Name
 1. `github_account_name <https://github.com/github_account_name>`__ for
    `Project subject <github_account_name/README.rst>`__:
    https://github.com/github_account_name/project_name
+2. `Yojahuang <https://github.com/Yojahuang>`__ for
+   `R tree search engine <Yojahuang/README.rst>`__:
```
Comment (yungyuc) [inline project/README.rst:72]: The file name should be `README.md`

**Review by yungyuc — APPROVED** [review]
You'd better add some amendments to the draft but in general it looks good to me.

Let me know when you want it to be merged.

File: project/Yojahuang/README.md (hunk 5 above)
Comment (Yojahuang) [reply project/Yojahuang/README.md:66]: NP, I'll put this task to week 3

**Comment by Yojahuang** [comment]
@yungyuc the PR is ready!

**Pull request merged**